### PR TITLE
NUTCH-2442 Injector to stop if job fails to avoid loss of CrawlDb

### PR DESCRIPTION
**Pull request by Omkar20895**
- Added Job status checks in the classes: Injector, ReadHostDb, CrawlCompletionStats, ProtocolStatusStatistics, SitemapProcessor and DomainStatistics.

### CODE REVIEW

File: src/java/org/apache/nutch/util/ProtocolStatusStatistics.java [hunk 1]
```diff
@@ -122,8 +122,17 @@ public int run(String[] args) throws Exception {
     job.setNumReduceTasks(numOfReducers);
 
     try {
-      job.waitForCompletion(true);
-    } catch (Exception e) {
+      boolean success = job.waitForCompletion(true);
+      if(!success){
```
Comment (sebastian-nagel) [inline src/java/org/apache/nutch/util/ProtocolStatusStatistics.java]: Please use consistent and uniform formatting.

Comment (Omkar20895) [reply src/java/org/apache/nutch/util/ProtocolStatusStatistics.java]: @sebastian-nagel I did not understand, the formatting looks good to me. Can you please elaborate on what I am missing here? Thanks.

Comment (sebastian-nagel) [reply src/java/org/apache/nutch/util/ProtocolStatusStatistics.java]: There is a missing space before opening brackets/braces (also other occurrences). The formatting should follow the style defined by the Eclipse Code Formatter rules (https://github.com/apache/nutch/blob/master/eclipse-codeformat.xml). The easiest way is to import the rules into Eclipse (and other IDEs), in doubt, Eclipse allows to format the code from the command-line.

Comment (Omkar20895) [reply src/java/org/apache/nutch/util/ProtocolStatusStatistics.java]: Apologies for such a silly query 😅 . I use vim editor and everything lgtm, but, after importing the project into eclipse I could see the differences in the format. Thanks.

**Comment by kpm1985** [comment]
This is somewhat of a tangent, but if we mavenize the build, we could use
some nice plugins for maven. Saw a few emails this week on this topic.

I looked into using nailgun and some other sweet things to ensure
consistency on a per commit basis, ie the maven import ordering plugin ( in
use under ASFLv2 in Fluo) for Log4J2.

Obviously, having each developer care enough to learn to import the style
into eclipse is good, a standard imo, but in the future we can automate
these styles. This is fantastic for formatting older class files etc..

Also, been lagging have lots of work but am serious about fixing old PR
(fyi for Sebastian)

Afaik ant is preferred less to maven.

**Comment by sebastian-nagel** [comment]
Yes, that's clear. But the initial challenge is to port the complex build system to Maven.

**Pull request merged**